akka-http-core/src/main/java/akka/http/javadsl/model/headers/HttpChallenge.java

```diff
@@ -15,17 +16,9 @@
     public abstract Map<String, String> getParams();
 
     public static HttpChallenge create(String scheme, String realm) {
-        return new akka.http.scaladsl.model.headers.HttpChallenge(scheme, realm, Util.emptyMap);
+        return akka.http.scaladsl.model.headers.HttpChallenge.apply(scheme, Option.apply(realm));
     }
     public static HttpChallenge create(String scheme, String realm, Map<String, String> params) {
-        return new akka.http.scaladsl.model.headers.HttpChallenge(scheme, realm, Util.convertMapToScala(params));
-    }
-
-    public static HttpChallenge createBasic(String realm) {
-        return create("Basic", realm);
-    }
```

Comment:
@ktoso Can you take another look, I followed @2beaucoup's  advice and added a constructor on the Java API for an optional realm. I also added some deprecation annotations to both the Java and Scala constructors to clarify which to use. Also added a FIXME to clean up the deprecated constructors in 3.0 - is that over the top? I also updated the tests and tested with out codebase to ensure that this PR really does make it possible to support browser based Kerberos/SPNEGO authentication (and it does)